Zend/Optimizer/compact_literals.c

```diff
@@ -820,23 +820,19 @@ void zend_optimizer_compact_literals(zend_op_array *op_array, zend_optimizer_ctx
 		zend_hash_destroy(&hash);
 		zend_arena_release(&ctx->arena, checkpoint);
 
-		if (1) {
-			opline = op_array->opcodes;
-			while (1) {
-				if (opline->opcode == ZEND_RECV_INIT) {
-					zval *val = &op_array->literals[opline->op2.constant];
+		opline = op_array->opcodes;
+		while (opline->opcode && opline->opcode != ZEND_RECV) {
```

Comment:
I don't understand this change. I feel like it should be a do-while loop. Otherwise you might never execute it even once.  Also, should it not be `while (opline->opcode && (opline->opcode == ZEND_RECV || opline->opcode == ZEND_RECV_INIT)) {`?